lib/tether/tether_linux.go

```diff
@@ -140,9 +140,8 @@ func (t *tether) stopReaper() {
 	signal.Reset(syscall.SIGCHLD)
 
 	// just closing the incoming channel is not going to stop the iteration
-	// so we use done channel to signal it
-	log.Debugf("Signalling the child reaper loop")
-	close(t.done)
+	// so we use the context cancellation to signal it
+	t.cancel()
```

Comment:
`t.cancel()` seems to indicate that the tether is done. I just thought it would be more flexible if each of these operations had their own channel to stop them executing. Its conceptually also clearer; may make this code easier to understand.